rbc/irtools.py

```diff
@@ -356,84 +375,85 @@ def compile_to_LLVM(functions_and_signatures,
     """
     target_desc = registry.cpu_target
 
-    if target is None:
+    if target_info is None:
         # RemoteJIT
-        target = TargetInfo.host()
+        target_info = TargetInfo.host()
         typing_context = target_desc.typing_context
         target_context = target_desc.target_context
     else:
         # OmnisciDB target
-        if target.is_cpu:
+        if target_info.is_cpu:
             typing_context = typing.Context()
-            target_context = RemoteCPUContext(typing_context, target)
-        elif target.is_gpu:
-            typing_context = RemoteGPUTypingContext()
-            target_context = RemoteGPUTargetContext(typing_context, target)
+            target_context = JITRemoteCPUContext(typing_context)
+        elif target_info.is_gpu:
+            typing_context = JITRemoteGPUTypingContext()
+            target_context = JITRemoteGPUTargetContext(typing_context)
```

Comment:
It looks like we can eliminate `target_info.is_gpu` block here and merge `JITRemoteCPUContext` and `JITRemoteGPUTargetContext` to `JITRemoteTargetContext` to simplify irtools.py code a little.
After all, for GPU LLVM IR we are using numba CPU code generation with minor adjustments.

Reply:
Actually, we need the `if-else` block to map math functions to libdevice when the target has GPU enabled. See:
https://github.com/xnd-project/rbc/blob/a34604c848250b670a79b214effffb8546ccdc7a/rbc/irtools.py#L182-L189